infra/binder/src/main/java/org/apache/shardingsphere/infra/binder/decider/engine/SQLFederationDeciderEngine.java

```diff
@@ -62,6 +62,9 @@ public SQLFederationDeciderContext decide(final QueryContext queryContext, final
         SQLFederationDeciderContext result = new SQLFederationDeciderContext();
         SQLStatementContext<?> sqlStatementContext = queryContext.getSqlStatementContext();
         // TODO move this logic to SQLFederationDecider implement class when we remove sqlFederationEnabled
+        if (queryContext.getSql().contains("WHEN")) {
```

Comment:
Why add this judgement?

Reply:
The "WHEN" is not supported by the federation query.